### PR TITLE
Replace deprecated cx_hash

### PR DESCRIPTION
**Pull request by orhoj**
## Purpose
To make the build pipeline run green again. Ledger has updated their SDKs and in those updates they have, among other things, deprecated a number of functions that we are using (e.g. for hashing). At the same time some font changes have been introduced for the Nano X meaning that the end to end tests also had to be updated accordingly.

## Changes
- Update end2end test screenshots.
- Use `no_throw` Ledger SDK functions and then add a check that the return code is `CX_OK`. If not, then we throw an error to keep functionality close to how it was prior to the SDK update by Ledger. In a future update we can make additional adjustments if need be.
- Fixed a NANO SP test that ran on the Nano X emulated device.

## Checklist
- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: src/util.c [hunk 1]
```diff
@@ -143,23 +143,40 @@ void getIdentityAccountDisplay(uint8_t *dst, size_t dstLength, uint32_t identity
...
+    // mimic the old library functions that would do a similar throw.
+
+    if (errorCode != CX_OK) {
+        THROW(ERROR_FAILED_CX_OPERATION);
+    }
+}
+
 void getPrivateKey(uint32_t *keyPathInput, uint8_t keyPathLength, cx_ecfp_private_key_t *privateKey) {
-    uint8_t privateKeyData[32];
+    // TODO Why does it require 64 bytes now instead of 32?
```
Comment (shjortConcordium) [inline src/util.c:163]: When would we resolve this mystery?

File: src/util.h [hunk 2]
```diff
@@ -91,6 +91,31 @@ int parseKeyDerivationPath(uint8_t *cdata);
  */
 void sign(uint8_t *input, uint8_t *signatureOnInput);
 
+/**
+ * Performs the hashing operation. This is our internal replacement of the deprecated cx_hash
+ * provided by Ledger. The function throws an ERROR_FAILED_HASHING error if the hashing fails.
```
Comment (shjortConcordium) [inline src/util.h:96]: what is `ERROR_FAILED_HASHING`? (Doesn't seem to be mentioned/defined anywhere else)

**Review by shjortConcordium — APPROVED** [review]

File: src/util.h (hunk 2 above)
Comment (orhoj) [reply src/util.h:96]: Old name. Changed to `ERROR_FAILED_CX_OPERATION`.

File: src/util.c (hunk 1 above)
Comment (orhoj) [reply src/util.c:163]: Going down the callstack I hit a `SYSCALL`, and I don't know if we can see what they actually do.

Comment (shjortConcordium) [reply src/util.c]: As far as I can see, the first 32 bytes are the key, and the remaining is the "chain code" used to derive further down the path.
https://github.com/satoshilabs/slips/blob/master/slip-0010.md

Comment (orhoj) [reply src/util.c]: I think you are right, I meant more like seeing their actual implementation to be absoutely certain. I think we're safe as all the tests work and they use the derived keys.

**Pull request merged**